core/state_processor.go

```diff
@@ -195,6 +257,28 @@ func applyTransaction(msg types.Message, config *params.ChainConfig, bc ChainCon
 	return receipt, err
 }
 
+// Apply State
+func (p *StateProcessor) Apply(block *types.Block) error {
+	// Process our block and retrieve external blocks.
+	receipts, _, statedb, usedGas, err := p.Process(block)
```

Comment:
How do we know if a block can be applied, without knowing the state of after the prior block?